### PR TITLE
optimize seed:dsls

### PR DESCRIPTION
**Pull request by wjordan**
This PR writes an md5 content-hash to DSL-defined levels to allow us to skip previously-seeded identical DSL-defined level content during incremental re-seeding.

Consider this a followup to #23805 for general dashboard-seed performance improvements.

## Timing changes on my development machine

Using command `time bundle exec rake seed:dsls`

**Before** (at e586d2d11b65311fe81fca0ccce543e70ae71484):
- Incremental seed (no changes, average of three trials): 0:49.83 (`seed:dsls` task 0:35.83)

**After** (at b1190df74234300d2617d503f33f8d389524ce01):
- Incremental seed (no changes, average of three trials): 0:14.35 (`seed:dsls` task 0:01.43)

### CODE REVIEW

**Review by islemaster — APPROVED** [review]
Looks great!  Tested seamlessly on my machine.

File: dashboard/app/dsl/base_dsl.rb [hunk 1]
```diff
@@ -32,7 +32,7 @@ def self.parse(str, filename, name=nil)
     object = new
     object.name(name) if name.present?
     object.instance_eval(str.to_ascii, filename)
-    [object.parse_output, object.i18n_hash]
+    [object.parse_output.tap {|out| out[:md5] = Digest::MD5.hexdigest(str)}, object.i18n_hash]
```
Comment (joshlory) [inline dashboard/app/dsl/base_dsl.rb:35]: Nit: clearer without `tap`?

```suggestion
    out = object.parse_output
    out[:md5] = Digest::MD5.hexdigest(str)
    [out, object.i18n_hash]
```

**Review by joshlory — COMMENTED** [review]
This is cool! The seed time speed-up numbers look great.

**Comment by joshlory** [comment]
Worth de-duping this code against the similar method in non-DSL level types?

https://github.com/code-dot-org/code-dot-org/blob/8990f16a29d8a1cf9dba0ffb220bd424e8542810/dashboard/lib/level_loader.rb#L87-L95

**Comment by wjordan** [comment]
CI test failures are relevant, need to debug/fix them before this can be merged.

**Pull request closed without merging**